ToDoListAPI/ToDoListAPI/Services/TaskService.cs

```diff
@@ -0,0 +1,115 @@
+﻿using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using ToDoListAPI.IService;
+using ToDoListAPI.Models;
+
+namespace ToDoListAPI.Services
+{
+    public class TaskService : ITaskService
+    {
+        private MyDBContext _context;
+        private readonly ILogger<TaskService> _logger;
+
+        public TaskService(MyDBContext context, ILogger<TaskService> logger)
+        {
+            _context = context;
+            _logger = logger;
+        }
+
+        public async Task<List<TaskRequest>> GetTask(int userId, bool status, DateTime date)
+        {
+            //return await _context.Tasks.ToListAsync();
+            _logger.LogInformation("get all task");
+            if (status != null && date == null)
+            {
+                return await _context.Tasks.Where(x => x.UserId == userId && x.Status == status).Select(x => taskDTO(x)).ToListAsync();
+            }
+            if (status == null && date != null)
+            {
+                return await _context.Tasks.Where(x => x.UserId == userId && x.CreateAt == date).Select(x => taskDTO(x)).ToListAsync();
+            }
+            return await _context.Tasks.Where(x => x.UserId == userId && x.CreateAt == date && x.Status == status).Select(x => taskDTO(x)).ToListAsync();
```

Comment:
```suggestion
            if (status != null && date == null)
            {
                return await _context.Tasks.Where(x => x.UserId == userId && x.Status == status).Select(x => taskDTO(x)).ToListAsync();
            }
            if (status == null && date != null)
            {
                return await _context.Tasks.Where(x => x.UserId == userId && x.CreateAt == date).Select(x => taskDTO(x)).ToListAsync();
            }
            return await _context.Tasks.Where(x => x.UserId == userId && x.CreateAt == date && x.Status == status).Select(x => taskDTO(x)).ToListAsync();
```
cần viết lại query chỗ này nhé